core/templates/dev/head/App.ts

```diff
@@ -130,20 +130,25 @@ angular.module('oppia').config([
             return config;
           },
           responseError: function(rejection) {
+            var warningMessage = 'Error communicating with server.';
             // A rejection status of -1 seems to indicate (it's hard to find
             // documentation) that the response has not completed,
             // which can occur if the user navigates away from the page
             // while the response is pending, This should not be considered
             // an error.
             if (rejection.status !== -1) {
               $log.error(rejection.data);
-
-              var warningMessage = 'Error communicating with server.';
               if (rejection.data && rejection.data.error) {
                 warningMessage = rejection.data.error;
               }
               AlertsService.addWarning(warningMessage);
             }
+            var rejectionUrl = typeof rejection.config !== 'undefined' ? (
```

Comment:
> I've added checks for this. Done.

Um -- actually, I meant, could you check the API/docs to understand what is guaranteed to be passed? We should not be adding spurious undefined checks either. If rejection.config is always present then we should not add the "is undefined" check.

Reply:
Yes, I could not find any documentation that says rejection.config.url would be available. I think the rejection object is a rejected promise. So if the promise does not contain a config object it will be undefined.

Additionally, the $http method documentation on [this page](https://docs.angularjs.org/api/ng/service/$http#post), lists `config` as an optional parameter. So it is possible that it may be undefined, I think.

Reply:
Sounds good, the "optional parameter" is exactly what to look for here. Please add a comment stating that the "config" parameter is optional with a link to that page. Thanks!